igrill.py

```diff
@@ -110,6 +113,9 @@ def authenticate(self):
 
     def read_battery(self):
         return float(bytearray(self.battery_char.read())[0])
+    
+    def read_heating_elements(self):
```

Comment:
if we add heating_element to IDevicePeripheral.init as I suggest below, we can wrap this in an if statement, and return None if it is not present on the device type. This will make it easy to keep the publish code generic.